src/test/java/org/javarosa/core/model/SelectOneChoiceFilterTest.java

```diff
@@ -131,6 +131,7 @@ public void clearingValueAtLevel1_ShouldClearValuesAtLevels2And3() {
         // If we set level2 to "aa", form validation passes. Currently, clearing a choice only updates filter expressions
         // that directly depend on it. With this form, we could force clearing the third level when the first level is cleared
         // by making the level3 filter expression in the form definition reference level1 AND level2.
+        scenario.answer("/data/level1", "b");
```

Comment:
A side-effect of this change is that it's no longer possible to set a value for the second level of a cascade if the first level isn't set. That's a good thing but in practice, I don't think it will make any difference for users since they'd have to visit the first level in order to change its value.